File: shadowed-config-codecs/src/main/java/io/github/legacy_fabric_community/shadowedconfig/codecs/ops/ExtendedOps.java
package io.github.legacy_fabric_community.shadowedconfig.codecs.ops;

import java.util.function.Consumer;

import com.mojang.serialization.DataResult;
import com.mojang.serialization.DynamicOps;
import io.github.legacy_fabric_community.shadowedconfig.codecs.SetBuilder;

public interface ExtendedOps<T> extends DynamicOps<T> {
	default SetBuilder<T> setBuilder() {
		return new SetBuilder.Builder<>(this);
	}

	default DataResult<Consumer<Consumer<T>>> getSet(T input) {
		return this.getList(input);
	}
}

File: shadowed-config-codecs/src/main/scala/io/github/legacy_fabric_community/shadowedconfig/codecs/ops/ExtendedOps.scala
package io.github.legacy_fabric_community.shadowedconfig.codecs.ops

import java.util.function.Consumer

import com.mojang.serialization.DataResult
import com.mojang.serialization.DynamicOps
import io.github.legacy_fabric_community.shadowedconfig.codecs.{SetBuilder, ImmutableSetBuilder}


trait ExtendedOps[T] extends DynamicOps[T] {
	def setBuilder = new ImmutableSetBuilder[T](this)

	def getSet(input: T): DataResult[Consumer[Consumer[T]]] = this.getList(input)
}
